### PR TITLE
Feature/02 profile json

### PR DESCRIPTION
**Pull request by longnghiem**

### CODE REVIEW

File: __tests__/profiles.test.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/* eslint-disable */
```
Comment (ximbong91023) [inline __tests__/profiles.test.js]: Enable it afterward :)

File: app.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ const path = require("path");
 const cookieParser = require("cookie-parser");
 const logger = require("morgan");
 const mongoose = require("mongoose");
+const bodyParser = require("body-parser");
```
Comment (ximbong91023) [inline app.js]: Shall we use express's own bodyparser instead

File: routes/profiles.js [hunk 3]
```diff
@@ -1,18 +1,41 @@
...
 
+// show all profiles
 router.get("/", (req, res) => {
 	res.send(profiles);
 });
 
+// show form to edit 1 profile
+router.get("/:id/edit", (req, res) => {
+	const id = req.params.id;
+	const targetedProfile = profiles.filter((profile) => profile._id === id)[0];
```
Comment (ximbong91023) [inline routes/profiles.js]: Use find instead

File: routes/profiles.js [hunk 4]
```diff
@@ -1,18 +1,41 @@
...
+});
+
+// show profile of 1 user
 router.get("/:id", (req, res) => {
 	const id = req.params.id;
-	const targeted_profile = profiles.filter((profile) => {
-		return profile._id === id;
-	});
-	res.send(targeted_profile);
+	const targetedProfile = profiles.filter((profile) => profile._id === id)[0];
```
Comment (ximbong91023) [inline routes/profiles.js]: Use find instead.

File: routes/profiles_functions/profiles_functions.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+		if (profile._id === id) {
+			profile = { ...profile, ...new_data, };
+		}
+		return profile;
+	});
+	const editedProfile = profiles.filter((profile) => profile._id === id)[0];
+	return editedProfile;
+};
+
+module.exports.deleteProfile = deleteProfile;
```
Comment (ximbong91023) [inline routes/profiles_functions/profiles_functions.js]: module.exports = {deleteProfile, deleteProfile}; seems better

File: routes/profiles.js [hunk 6]
```diff
@@ -1,18 +1,41 @@
...
+router.get("/:id/edit", (req, res) => {
+	const id = req.params.id;
+	const targetedProfile = profiles.filter((profile) => profile._id === id)[0];
+	res.send(`Edit Profile Form for user ${targetedProfile.firstName} ${targetedProfile.lastName}`);
+});
+
+// edit 1 profile
+router.put("/:id", (req, res) => {
+	const id = req.params.id;
+	const new_data = req.body;
```
Comment (ximbong91023) [inline routes/profiles.js]: camelCase variable!

**Comment by longnghiem** [comment]
Changes were made, please review my changes.

File: routes/profiles_functions/profiles_functions.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
+const deleteProfile = (id, profiles) => {
+	profiles = profiles.filter((profile) => profile._id !== id);
+	return profiles;
+};
+
+const editProfile = (id, new_data, profiles) => {
```
Comment (ximbong91023) [inline routes/profiles_functions/profiles_functions.js]: Rewrite this function

**Comment by longnghiem** [comment]
Changes were made, please review my changes.

**Pull request merged**